README
Inline Text Editing [VERSION]
[DATE]
Corey Hart @ http://www.codenothing.com


changelog
###
### Inline Text Editing 1.3 Change Log
### April 26, 2010
### Corey Hart @ http://www.codenothing.com
###

# New Features
-Added revert hook to revert the current text back to it's starting position at first view
-Added postFormat setting to control post data sent to the server on each request
-Enhanced load functionality, can now control the data recieved from the server and do custom insertion
-Can now control the type of request sent to the server (GET or POST)
-Now preventing multiple inlineEdit initiations on the same wrapper element.

# Possible Breaks
-load function has changed completely, it gets passed 2 arguments, the event object that triggered the request,
and a ui object holding the response and settings.


jquery.inline-edit.pack.js
/**
 * In-Line Text Editing
 * April 3, 2008
 * Corey Hart @ http://www.codenothing.com
 *
 * @href: Link to script for ajax call
 * @display: Displayed text grouping
 * @form: Form grouping
 * @text: Text input element (INPUT or TEXTAREA)
 * @save: Save Button
 * @cancel: Cancel Button
 * @loadtxt: html to replace display text with while ajax call completes
 * @hover: Class for mouseover/mouseout's on the display grouping
 */ 
;(function(a){a.fn.inlineEdit=function(b){var d;var c=a.extend({href:"ajax.php",display:".display",form:".form",text:".text",save:".save",cancel:".cancel",loadtxt:"Loading...",hover:"none-error-404"},b||{});this.each(function(){a(c.display,this).mouseover(function(){a(this).addClass(c.hover);}).mouseout(function(){a(this).removeClass(c.hover);}).click(function(){d=a(this).html();a(this).hide().siblings(c.form).show().find(c.text).val(d).focus();return false;});a(c.cancel,this).click(function(){a(this).parents(c.form).hide().siblings(c.display).show();return false;});a(c.save,this).click(function(){d=a(this).parents(c.form).find(c.text).val();a(this).parents(c.form).hide().siblings(c.display).html(c.loadtxt).load(c.href,{text:d}).show();return false;});});};})(jQuery);


src/jquery.inline-edit.goog.js
/*
 * Inline Text Editing 1.3
 * April 26, 2010
 * Corey Hart @ http://www.codenothing.com
 */ 
(function(d,f){d.fn.inlineEdit=function(m){return this.each(function(){var i=this,g=d(i),e,a=d.extend({href:"ajax.php",requestType:"POST",html:true,load:f,display:".display",form:".form",text:".text",save:".save",cancel:".cancel",revert:".revert",loadtxt:"Loading...",hover:f,postVar:"text",postData:{},postFormat:f},m||{},d.metadata?g.metadata():{}),b=g.find(a.display),c=g.find(a.form),h=c.find(a.text),j=c.find(a.save),n=c.find(a.revert),o=c.find(a.cancel);if(d.data(i,"inline-edit")!==true){d.data(i,
"inline-edit",true);c.bind("submit.inline-edit",function(){j.trigger("click.inline-edit");return false});b.bind("click.inline-edit",function(){b.hide();c.show();if(a.html){if(e===f)e=b.html();h.val(e).focus()}else if(e===f)e=h.val();return false}).bind("mouseenter.inline-edit",function(){b.addClass(a.hover)}).bind("mouseleave.inline-edit",function(){b.removeClass(a.hover)});n.bind("click.inline-edit",function(){h.val(e||"").focus();return false});o.bind("click.inline-edit",function(){c.hide();b.show();
b.hasClass(a.hover)&&b.removeClass(a.hover);return false});j.bind("click.inline-edit",function(k){a.postData[a.postVar]=h.val();c.hide();b.html(a.loadtxt).show();b.hasClass(a.hover)&&b.removeClass(a.hover);d.ajax({url:a.href,type:a.requestType,data:a.postFormat?a.postFormat.call(g,k,{settings:a,postData:a.postData}):a.postData,success:function(l){e=f;a.load?a.load.call(b,k,{response:l,settings:a}):b.html(l)}});return false})}})}})(jQuery);


src/jquery.inline-edit.js
/*
 * Inline Text Editing 1.3
 * April 26, 2010
 * Corey Hart @ http://www.codenothing.com
 */ 
(function( $, undefined ){

	$.fn.inlineEdit = function( options ) {
		return this.each(function(){
			// Settings and local cache
			var self = this, $main = $( self ), original,
				settings = $.extend({
					href: 'ajax.php',
					requestType: 'POST',
					html: true,
					load: undefined,
					display: '.display',
					form: '.form',
					text: '.text',
					save: '.save',
					cancel: '.cancel',
					revert: '.revert',
					loadtxt: 'Loading...',
					hover: undefined,
					postVar: 'text',
					postData: {},
					postFormat: undefined
				}, options || {}, $.metadata ? $main.metadata() : {} ),

				// Cache All Selectors
				$display = $main.find( settings.display ),
				$form = $main.find( settings.form ),
				$text = $form.find( settings.text ),
				$save = $form.find( settings.save ),
				$revert = $form.find( settings.revert ),
				$cancel = $form.find( settings.cancel );

			// Make sure the plugin only get initialized once
			if ( $.data( self, 'inline-edit' ) === true ) {
				return;
			}
			$.data( self, 'inline-edit', true );

			// Prevent sending form submission
			$form.bind( 'submit.inline-edit', function(){
				$save.trigger( 'click.inline-edit' );
				return false;
			});
	
			// Display Actions
			$display.bind( 'click.inline-edit', function(){
				$display.hide();
				$form.show();

				if ( settings.html ) {
					if ( original === undefined ) {
						original = $display.html();
					}

					$text.val( original ).focus();
				}
				else if ( original === undefined ) {
					original = $text.val();
				}

				return false;
			})
			.bind( 'mouseenter.inline-edit', function(){
				$display.addClass( settings.hover );
			})
			.bind( 'mouseleave.inline-edit', function(){
				$display.removeClass( settings.hover );
			});

			// Add revert handler
			$revert.bind( 'click.inline-edit', function(){
				$text.val( original || '' ).focus();
				return false;
			});

			// Cancel Actions
			$cancel.bind( 'click.inline-edit', function(){
				$form.hide();
				$display.show();

				// Remove hover action if stalled
				if ( $display.hasClass( settings.hover ) ) {
					$display.removeClass( settings.hover );
				}

				return false;
			});

			// Save Actions
			$save.bind( 'click.inline-edit', function( event ) {
				settings.postData[ settings.postVar ] = $text.val();
				$form.hide();
				$display.html( settings.loadtxt ).show();

				if ( $display.hasClass( settings.hover ) ) {
					$display.removeClass( settings.hover );
				}

				$.ajax({
					url: settings.href,
					type: settings.requestType,
					data: settings.postFormat ? 
						settings.postFormat.call( $main, event, { settings: settings, postData: settings.postData } ) :
						settings.postData,
					success: function( response ){
						original = undefined;

						if ( settings.load ) {
							settings.load.call( $display, event, { response: response, settings: settings } );
							return;
						}

						$display.html( response );
					}
				});

				return false;
			});
		});
	};

})( jQuery );
